signer/src/bitcoin/utxo.rs

```diff
@@ -175,23 +175,30 @@ impl SbtcRequests {
         // Filter deposit requests based on two constraints:
         // 1. The user's max fee must be >= our minimum required fee for deposits
         //     (based on fixed deposit tx size)
-        // 2. The deposit amount must be less than the per-deposit limit
-        // 3. The total amount being minted must stay under the maximum allowed mintable amount
+        // 2. The deposit amount must be greater than or equal to the per-deposit minimum
+        // 3. The deposit amount must be less than or equal to the per-deposit cap
+        // 4. The total amount being minted must stay under the maximum allowed mintable amount
         let minimum_deposit_fee = self.compute_minimum_fee(SOLO_DEPOSIT_TX_VSIZE);
         let max_mintable_cap = self.sbtc_limits.max_mintable_cap().to_sat();
         let per_deposit_cap = self.sbtc_limits.per_deposit_cap().to_sat();
+        let per_deposit_minimum = self.sbtc_limits.per_deposit_minimum().to_sat();
 
         let mut amount_to_mint: u64 = 0;
         let deposits = self.deposits.iter().filter_map(|req| {
             let is_fee_valid = req.max_fee.min(req.amount) >= minimum_deposit_fee;
+            let is_above_per_deposit_minimum = req.amount >= per_deposit_minimum;
```

Comment:
I think that this closure has gotten complex enough where we should break it out into a separate function or create some struct with some function. I feel that doing so would be much easier to understand and test.

Reply:
Done in https://github.com/stacks-network/sbtc/pull/1124/commits/555ffbaec8d59796bac43e4a72bb8cf3be602a43